packages/next/src/server/app-render/app-render.tsx

```diff
@@ -720,16 +721,22 @@ async function renderToHTMLOrFlightImpl(
     isRSCRequest &&
     req.headers[NEXT_ROUTER_PREFETCH_HEADER.toLowerCase()] !== undefined
 
+  // TODO: this should be moved to a higher point in the stack
```

Comment:
TODO nit: I think ideally we can just remove the flight router state dependency for interception routes. It's only needed right now for the `makeGetDynamicParams` function, but seems like with some refactoring we could get it to behave.